hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -1007,7 +975,7 @@ extends Importer:
           case R(id) =>
             val sym = members.getOrElse(id.name, die)
             val owner = ctx.outer.inner
-            val isModMember = owner.exists(_.isInstanceOf[ModuleSymbol])
+            val isModMember = owner.forall(_.isInstanceOf[ModuleSymbol]) // forall because a top-level def is also a module method
```

Comment:
Actually, this also includes nested functions, not only top-level ones.

```fs
fun main() =
  fun f(module m: M): module M = m
  f(M).x
```

Indeed, this should also work. But `f` is definitely not a "module method".

So maybe it'd make more sense to distinguish normal functions (within modules or without) from "methods" aka "object methods", which are the ones that are meant to be limited.

Reply:
Perhaps we should even use a different keyword for object methods, at this point, such as `method`...